src/weapon/w_050.c

```diff
@@ -486,7 +486,122 @@ void func_ptr_8017000C(Entity* self) {
     }
 }
 
-INCLUDE_ASM("weapon/nonmatchings/w_050", func_ptr_80170010);
+s32 func_ptr_80170010(Entity* self) {
+    const int PrimCount = 8;
+    Primitive* prim;
+    Point16 point;
+    s32 params;
+    s32 i;
+    s16 x, y;
+    s16 baseX, baseY;
+    s16 prim_x, prim_y;
+    s32 result;
+
+    if (PLAYER.ext.player.anim < 0x41 || PLAYER.ext.player.anim > 0x47 ||
+        !g_Player.unk46) {
+        DestroyEntity(self);
+        return;
+    }
+
+#ifdef VERSION_PSP
+    params = (self->params & 0x7F00) >> 8;
+#else
+    params = ((u8)(self->params >> 8)) & 0x7F;
+#endif
+
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(PRIM_GT4, PrimCount);
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+
+        self->flags = FLAG_HAS_PRIMS | FLAG_UNK_40000 | FLAG_UNK_20000;
+        if (params == 1) {
+            func_162000_8017BA38(self, &point, true);
+        } else {
+            func_162000_8017BA38(self, &point, false);
+        }
+
+        baseX = point.x;
+        baseY = point.y;
+
+        if (PLAYER.drawFlags & FLAG_DRAW_ROTY) {
+            if ((PLAYER.ext.player.anim - 0x41) == 2 ||
+                (PLAYER.ext.player.anim - 0x41) == 3) {
```

Comment:
This would match PS1 without the subtraction, but only matches PSP with.

Reply:
Given it's checking for 0x41 above and returning seems to be deliberate to me and not an optimization